pkg/cloud/aws/services/ec2/subnets.go

```diff
@@ -202,6 +202,12 @@ func (s *Service) describeVpcSubnets() (v1alpha1.Subnets, error) {
 
 		// ... or if it has an internet route
 		rt := routeTables[*ec2sn.SubnetId]
+		if rt == nil {
+			// If there is no explicit association, subnet defaults to main route table as implicit association
+			if mainRt, ok := routeTables[mainRouteTableInVPCKey]; ok {
+				rt = mainRt
```

Comment:
nit: I don't think we need the `if` here, since the zero value (what we get back for a missing key) of a pointer is nil. I think we can just do 
```
if rt == nil {
    // If there is no explicit association, subnet defaults to main route table as implicit association
    rt = routeTables[mainRouteTableInVPCKey]
}
```

+1 on the comment, that's very helpful.

Reply:
Ah, good call. Will fix.